### ISSUE TITLE
Hidden no longer works in v5.0.0

### ISSUE THREAD

**Issue by esso23** [skclusive/Skclusive.Material.Component#67]
v5.0.0-rc2

This code worked with the previous version. Now it's never hidden no matter the size of the screen. Seems like a global issue with Hidden.

```html
<Hidden MediumUp Context="HiddenContext">
  <IconButton Class="@HiddenContext.Class" Color="@Color.Inherit" Edge="@IconButtonEdge.Start" Style="margin-right: 10px;" OnTouchEnd="HandleDrawerTouch" OnMouseUp="HandleDrawerClick">
    <MenuIcon />
  </IconButton>
</Hidden>
```

**Comment by skclusive** [comment]
will look into that. thanks for reporting.

**Comment by skclusive** [comment]
Hidden seems to work in Docs site. https://skclusive.github.io/Skclusive.Material.Docs/hidden (v5.0.0-rc2).

will check the above sample whether something else broken.

btw `<ThemeProvider />` on root level is required. which injected DI registered styles and scripts. Material components register their styles in DI.

**Comment by esso23** [comment]
Yea, I have a <ThemeProvider /> in my App. This is an app we have deployed to production to several customers with Skclusive.Material.Component Version=3.0.2-preview1 - with this version it works OK.

Now I'm trying to upgrade to 5.0.0-rc2 and so far I have changed this:
1. Removed CancelIcon in my code - apparently it was added to the library
2. Removed Skclusive.Material.Script usings
3. Changed `<ThemeProvider Theme="@Theme.Light" />` to `<ThemeProvider />`, because the compiler says the Theme parameter does not exist
4. Added `using Skclusive.Material.Core;` to Startup.cs
5. Updated some Microsoft packages to .NET 5 RC2 versions

**Comment by skclusive** [comment]
if possible can you share minimal repo to reproduce?

the change between 3.0.2-preview1 and 5.0.0-rc2 is Skclusive.Material.Script classes are extracted and moved to respective component classes and util classes moved to Skclusive.Script.DomHelpers.

**Comment by rhyskoedijk** [comment]
I've also experienced this in 2.0, so I don't think it is specific to 5.0.
You can make it work by declaring the `HiddenContext.Class` on an element that contains the button. e.g.

This **doesn't** work:
````
            <Hidden LargeUp Context="HiddenContext">
                <IconButton
                    Class="@HiddenContext.Class"
                    Color="@Color.Inherit"
                    OnClick="@OnSidebarToggle">
                    <MenuIcon />
                </IconButton>
            </Hidden>
````

This **does** work:
````
            <Hidden LargeUp Context="HiddenContext">
                <div class="@HiddenContext.Class">
                    <IconButton
                        Color="@Color.Inherit"
                        OnClick="@OnSidebarToggle">
                        <MenuIcon />
                    </IconButton>
                </div>
            </Hidden>
````

If you inspect the DOM, you'll see that the `IconButton-Root` class overrides `Hidden`, which prevents the element from hiding.
You could probably also fix this by having the `Hidden` classes use `display: none !important`.

![image](https://user-images.githubusercontent.com/123988/98751647-5dbd2980-2425-11eb-881a-654d1aaa5c49.png)

**Comment by skclusive** [comment]
@rhyskoedijk great. thanks for the investigation and details.

yes. as hidden is critical we can make is display: none !important. or include the HiddenStyle above the button style to get higher priority by default.

**Comment by esso23** [comment]
@rhyskoedijk that actually helped, thanks (I was using it to wrap a `Fab` and a `Button`). Still weird that is worked fine in previous version.

**Comment by skclusive** [comment]
@esso23 it could be due to the order of style added then and now registration in DI. 
will add priority property for the style providers and built-in providers with priority slot, so custom style provider can hook in between built-in providers.

**Comment by skclusive** [comment]
issue is fixed in 5.1.0 release. Hidden class can be added to the Button directly without wrapper div. Hidden styles are added later to get more css specificity.
